### PR TITLE
fix: Sync typescript unlock condition types

### PR DESCRIPTION
**Pull request by marc2332**
Sync typescript unlock condition types.

Rust uses `slot` not `slotIndex`, you can see this for instance in `TimelockUnlockConditionDto`.

### CODE REVIEW

**Review by Thoralf-M — CHANGES_REQUESTED** [review]
Please add a changelog entry

**Comment by marc2332** [comment]
> Please add a changelog entry

Done, did I do it correctly? :eyes:

File: bindings/nodejs/CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Security -->
 
+## Unreleased - 2024-04-04
+
+### Fixed
+
+- Updated the TypeScript types for Unlock Conditions to match their Rust equivalent.;
```
Comment (Thoralf-M) [inline bindings/nodejs/CHANGELOG.md]: Not wrong, but I would rather do something like this
```suggestion
## 2.0.0-alpha.4 - 2024-04-DD

### Changed

- Renamed `{TimelockUnlockCondition, ExpirationUnlockCondition}::slotIndex to slot`;
```

Comment (marc2332) [reply bindings/nodejs/CHANGELOG.md]: yeah better

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**